tests/test_env_checker.py

```diff
@@ -0,0 +1,28 @@
+import gym
+import numpy as np
+from gym.spaces import Box, Dict, Discrete
+
+from stable_baselines3.common.env_checker import check_env
+
+
+class ActionDictTestEnv(gym.Env):
+    action_space = Dict({"position": Discrete(1), "velocity": Discrete(1)})
+    observation_space = Box(low=-1.0, high=2.0, shape=(3,), dtype=np.float32)
+
+    def step(self, action):
+        observation = np.array([1.0, 1.5, 0.5])
+        reward = 1
+        done = True
+        info = {}
+        return observation, reward, done, info
+
+    def reset(self):
+        return np.array([1.0, 1.5, 0.5])
+
+    def render(self, mode="human"):
+        pass
+
+
+def test_check_env_dict_action():
+    test_env = ActionDictTestEnv()
+    check_env(env=test_env, warn=True)
```

Comment:
This should test that a warning is indeed given (see e.g. [this](https://github.com/DLR-RM/stable-baselines3/blob/23afedb254d06cae97064ca2aaba94b811d5c793/tests/test_envs.py#L27))

Reply:
The point of the test is that a warning is not given because of my change. If i ran this test pre-change, the test fails.

Reply:
Yes this test checks for the original "nan" problem, but should also check that it warns about not-supported spaces. Warnings do not fail tests by default, but you can test if one is risen with

```python
with pytest.warns(Warning):
        check_env(env=test_env, warn=True)
```